pkg/sql/bulkmerge/merge_processor.go

```diff
@@ -156,8 +165,117 @@ func (m *bulkMergeProcessor) handleRow(row rowenc.EncDatumRow) (rowenc.EncDatumR
 
 // Start implements execinfra.RowSource.
 func (m *bulkMergeProcessor) Start(ctx context.Context) {
-	m.StartInternal(ctx, "bulkMergeProcessor")
-	m.input.Start(ctx)
+	ctx = m.StartInternal(ctx, "bulkMergeProcessor")
+	// TODO(annie): also find a way to process input/get task from coordinator
+	if err := m.mergeSSTs(ctx, 0); err != nil {
+		// TODO(annie): need to do this?
+		m.MoveToDraining(err)
+	}
+	m.MoveToDraining(nil)
+}
+
+func getKeysForTask(task int, splits []roachpb.Key) (roachpb.Key, roachpb.Key) {
+	if task == 0 {
+		return nil, splits[0]
+	}
+	last := len(splits) - 1
+	if task == last {
+		return splits[task], nil
+	}
+	if task > last {
+		panic(errors.AssertionFailedf("bad"))
+	}
+	return splits[task-1], splits[task]
+}
+
+// This is stolen from roachpb/data.go where we check if spans overlap.
+func isOverlapping(startKey, endKey roachpb.Key, sst execinfrapb.BulkMergeSpec_SST) bool {
+	otherStartKey, otherEndKey := sst.StartKey, sst.EndKey
+	if len(endKey) == 0 && len(otherEndKey) == 0 {
+		return startKey.Equal(otherStartKey)
+	} else if len(endKey) == 0 {
+		return bytes.Compare(startKey, otherStartKey) >= 0 && bytes.Compare(startKey, otherEndKey) < 0
+	} else if len(otherEndKey) == 0 {
+		return bytes.Compare(otherStartKey, startKey) >= 0 && bytes.Compare(otherStartKey, endKey) < 0
+	}
+	return bytes.Compare(endKey, otherStartKey) > 0 && bytes.Compare(startKey, otherEndKey) < 0
+}
+
+func (m *bulkMergeProcessor) mergeSSTs(ctx context.Context, taskID int) error {
+	// TODO(annie): get task from coordinator (look into how flows w multiple processors do it)
+	startKey, endKey := getKeysForTask(taskID, m.spec.Splits)
+	var storeFiles []storageccl.StoreFile
+	outputName := fmt.Sprintf("%s%d.sst", *m.spec.OutputUri, taskID)
+	dest, err := m.flowCtx.Cfg.ExternalStorageFromURI(ctx, outputName, username.RootUserName())
```

Comment:
The dest external storage should be opened by passing it m.spec.OutputUri. The SST writer you are using will create names that look like "data/<unique_name>.sst" and append it to the uri that is set on the external storage.